boost: Remove boost dependency

jdemel opened the pull request:
VOLK used Boost in the past quite heavily. We reduced Boost usage since then. At the moment, we only require `boost::filesystem` for very old systems. For all recent systems we rely on `std::filesystem` or `std::experimental::filesystem`.

The latest Ubuntu version that still required Boost was 14.04. Starting with Ubuntu 16.04, we can use `std::experimental::filesystem` and later `std::filesystem`.

Fix #174

jdemel committed:
boost: Remove boost dependency

VOLK used Boost in the past quite heavily. We reduced Boost usage since
then. At the moment, we only require `boost::filesystem` for very old
systems. For all recent systems we rely on `std::filesystem` or
`std::experimental::filesystem`.

The latest Ubuntu version that still required Boost was 14.04. Starting
with Ubuntu 16.04, we can use `std::experimental::filesystem` and later
`std::filesystem`.

Fix #174

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

jdemel committed:
boost: Update AppVeyor CI

AppVeyor CI still depends on ancient toolchains. This is an effort to
modernize the toolchain. It comes with the added benefit that we
simplify the whole configuration.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

jdemel committed:
appveyor: Update AppVeyor CI to VS 2019

AppVeyor fails to work correctly with AVX512 on old toolchains. We move
to a newer toolchain to fix this issue.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

jdemel commented:
So far, all CI tests pass without Boost. However, AppVeyor complains:
```
C:\projects\volk\kernels\volk/volk_32f_sin_32f.h(515): error C3861: '_mm512_floor_ps': identifier not found 
```
This is just an example. It seems like VS 2017 doesn't like to compile AVX512 because it doesn't support it although CMake reports its availability. Thus, we try to move to VS 2019 on AppVeyor as well.

marcusmueller commented:
> So far, all CI tests pass without Boost. However, AppVeyor complains:
> 
> ```
> C:\projects\volk\kernels\volk/volk_32f_sin_32f.h(515): error C3861: '_mm512_floor_ps': identifier not found 
> ```

> This is just an example. It seems like VS 2017 doesn't like to compile AVX512 because it doesn't support it although CMake reports its availability. Thus, we try to move to VS 2019 on AppVeyor as well.

This seems to be a separate issue and should be tracked separately, I guess?

jdemel commented:
> > So far, all CI tests pass without Boost. However, AppVeyor complains:
> > ```
> > C:\projects\volk\kernels\volk/volk_32f_sin_32f.h(515): error C3861: '_mm512_floor_ps': identifier not found 
> > ```
> 
> > This is just an example. It seems like VS 2017 doesn't like to compile AVX512 because it doesn't support it although CMake reports its availability. Thus, we try to move to VS 2019 on AppVeyor as well.
> 
> This seems to be a separate issue and should be tracked separately, I guess?

Yes, this is a separate issue.

michaelld reviewed (changes requested):
Top-level CMakeLists.txt:29 ... let's just change the CXX_STANDARD to 17 ... no reason to bother with 11 any longer!

michaelld commented on CMakeLists.txt:
```diff
@@ -195,13 +177,8 @@ if(${FILESYSTEM_FOUND})
     set(CMAKE_CXX_STANDARD 17)
```
Per my Request: Let's just move these up to the STANDARD definitions ; that simplifies the current `if/else` to just `if(NOT FILESYSTEM_FOUND` ... no need to `${}` around it either, BTW

jdemel replied:
Agreed. Fixed it.

jdemel committed:
c++: Require C++17 for std::filesystem

We rely on `std::filesystem` now, which came with C++17. Thus, we
require C++17 now.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

michaelld reviewed (approved):
LGTM
